internal/kubecmd/kubecmd.go

```diff
@@ -104,7 +104,8 @@ func (k *KubeCmd) UpgradeNodeVersion(ctx context.Context, conf *config.Config, f
 	provider := conf.GetProvider()
 	attestationVariant := conf.GetAttestationConfig().GetVariant()
 	region := conf.GetRegion()
-	imageReference, err := k.imageFetcher.FetchReference(ctx, provider, attestationVariant, conf.Image, region)
+	imageReference, err := k.imageFetcher.FetchReference(ctx, provider, attestationVariant,
```

Comment:
If I understand the code correctly, `imageReference` is a URI with a custom format designed by us.
How does this work for image upgrades / auto scaling if we set this URI as the image reference here?
I would think this would either require to do additional transformations on `imageReference` or changes in the node-operator.